src/app/shared/services/share/share.service.ts

```diff
@@ -30,90 +37,140 @@ export class ShareService extends SyncServiceBase {
 
   private registerTemplateActionHandlers() {
     this.templateActionRegistry.register({
-      share: async ({ args }) => {
-        const [actionId, ...shareArgs] = args;
-        const childActions = {
-          file: async () => await this.shareFile(shareArgs[0]),
-          text: () => this.share({ text: shareArgs[0] }),
-          url: () => this.share({ url: shareArgs[0] }),
-        };
-        // To support deprecated "share" action (previously used to share text only),
-        // assume text is being shared if first arg is not an actionId
-        if (!(actionId in childActions)) {
-          return await this.share({ text: args[0] });
+      share: async (action) => {
+        const { args, params } = action as { args: string[]; params: IShareActionParams };
+
+        if (params) {
+          await this.handleShare(params);
+        } else if (args) {
+          console.warn("[SHARE] Deprecated action syntax. Use `share | data_type: data` instead.");
+          await this.handleLegacyShare(args);
```

Comment:
nit(blocking): I think it would be better to map the legacy args to new params instead of triggering legacy methods if that's possible. This can be done outside of the if statement and then just have a check for params after